lib/si-layer-cache/src/hybrid_cache.rs

```diff
@@ -107,7 +107,12 @@ where
         let cache: HybridCache<Arc<str>, MaybeDeserialized<V>> = HybridCacheBuilder::new()
             .with_name(&config.name)
             .memory(memory_cache_capacity_bytes)
-            .with_weighter(|_key: &Arc<str>, value: &MaybeDeserialized<V>| size_of_val(value))
+            .with_weighter(
+                |_key: &Arc<str>, value: &MaybeDeserialized<V>| match value {
+                    MaybeDeserialized::RawBytes(bytes) => bytes.len(),
+                    MaybeDeserialized::DeserializedValue { size_hint, .. } => *size_hint,
+                },
+            )
```

Comment:
Any idea what the difference ends up being between the uncompressed postcard and the actual rust types? Feels like they can't be identical.

I wonder if a future solution to get it totally right is a SizeOf trait of some kind, where we can implement a function on every type that will report its size reliably